dask/dataframe/indexing.py

```diff
@@ -58,7 +58,16 @@ def __getitem__(self, key):
         if iindexer != slice(None):
             raise NotImplementedError(msg)
 
-        return self._iloc(iindexer, cindexer)
+        if len(set(self.obj.columns)) < len(self.obj.columns):
+            # if there are any duplicate column names, do an iloc
+            return self._iloc(iindexer, cindexer)
+        else:
+            # otherwise dispatch to dask.dataframe.core.DataFrame.__getitem__
+            return self._getitem(cindexer)
+
+    def _getitem(self, cindexer):
+        col_names = self.obj.columns[cindexer]
+        return self.obj.__getitem__(col_names)
```

Comment:
Thoughts on inlining this method?  If it's not going to be used elsewhere I wouldn't mind just sticking the two lines directly into the method above.

Reply:
happy to inline it, and it's unlikely to be used elsewhere.